### PR TITLE
FS-2125, FS-2126, BAU: Anchors for comments, Flag application validation

### PR DESCRIPTION
**Pull request by tferns**
Pull request includes changes for multiple things:

- Anchors for comments
    - I've refactored and taken the duplicate comment creation logic and put it at the top
    - We handle scenarios for initially expanding a comment, and submitting it
    - Having this at the top means we can return redirects when required, otherwise continue the flow without indentation

- Role checks for displaying Flags and "Flag application" button 
    - Now, only Assessor's and Lead Assessors can create & view flags
    - Also refactored the form to use govukTextarea rather than out own logic

Edit:  Have since pushed https://github.com/communitiesuk/funding-service-design-assessment/pull/107/commits/dc89424e10bc082bbaf52ceecd2c08bbe9f439ef to include some styling/qol changes, i.e. using proper error messages instead of hard-coding in the html template

### CODE REVIEW

File: app/assess/templates/macros/comments_summary.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {% macro comment_summary(comments, themes) %}
-<div class="govuk-!-margin-top-7" id="anchor">
+<div class="govuk-!-margin-top-7" id="comments">
```
Comment (harryyo) [inline app/assess/templates/macros/comments_summary.html:2]: Should this be "comment" (see above)?

**Review by harryyo — CHANGES_REQUESTED** [review]
Just small qs

File: app/assess/routes.py [hunk 2]
```diff
@@ -40,8 +40,53 @@ def display_sub_criteria(
     current_app.logger.info(f"Processing GET to {request.path}.")
     sub_criteria = get_sub_criteria(application_id, sub_criteria_id)
     theme_id = request.args.get("theme_id", sub_criteria.themes[0].id)
+    comment_form = CommentsForm()
+
+    add_comment_argument = request.args.get("add_comment") == "1"
+    if all(
+        [
+            add_comment_argument,
+            request.method == "POST",
```
[the remark this answers is not in the document]
Comment (tferns) [reply app/assess/routes.py:49]: Ah.  Good spot, I've changed to only use two conditions with `and`

File: app/assess/routes.py [hunk 3]
```diff
@@ -40,8 +40,53 @@ def display_sub_criteria(
...
+            url_for(
+                "assess_bp.display_sub_criteria",
+                application_id=application_id,
+                sub_criteria_id=sub_criteria_id,
+                theme_id=theme_id,
+                add_comment=1,
+                _anchor="comment",
+            )
+        )
+
```
[the remark this answers is not in the document]
Comment (tferns) [reply app/assess/routes.py:82]: Hmm.  I've just refactored this with how it already worked, this small bit of code is really only used for adding the initial "add_comment" query parameter which shows the add comment box allowing submissions.  The anchor is `comment` as it matches the id in the form used to write and submit a comment, so we anchor to that: 

https://github.com/communitiesuk/funding-service-design-assessment/blob/dc89424e10bc082bbaf52ceecd2c08bbe9f439ef/app/assess/templates/sub_criteria.html#L55-L69

File: app/assess/templates/macros/comments_summary.html (hunk 1 above)
Comment (tferns) [reply app/assess/templates/macros/comments_summary.html:2]: I think this is correct, there are two anchors, `comment` for the "add comment" box form, and "comments" for the actual displaying of comments

**Review by harryyo — APPROVED** [review]

**Pull request merged**